### ISSUE TITLE
Add the req, params, sid way of sending requests

### ISSUE THREAD

**Issue by the-marenga** [the-marenga/sf-api#96]
S&F stopped using encrypted requests and is now using a "cmd.php?" endpoint with "req, params, sid" params. The old method might get deprecated soon, so it is probably worth switching to that as well

UPDATE: For unknown reasons, they just switched back to their old format. No idea if they broke the mfbot, or some other tooling, but for the time being, this seems to be back to normal?

**Comment by the-marenga** [comment]
https://github.com/the-marenga/sf-api/issues/164#issuecomment-2703254249

**Comment by alexb231** [comment]
hey short question @the-marenga is this a planned thing ?

**Comment by the-marenga** [comment]
Oh, yes. Already implemented this in the bot and released that a few days ago in the current alpha. From what I can tell by the error logs, my implementation there worked without any issues, so I feel confident to implement that here as well soon. I mean it is basically just splitting the commands on ":" and removing the encryption. Should be easy to do here as well.

Any particular reason, why you are asking, or just cautious, that S&F might disable/ban the old approach?

**Comment by the-marenga** [comment]
Ok. I am working on rust S&F stuff right now anyways, so I might as well just start doing this here rn.

Btw. did you ever figure out your bot wasting mushrooms bug? I found something similar in the MFBot and it required a command not even implemented in my API here to fix. Something about the tower, a server bug and wrong update times. If that is still an issue I can explain further (& of course implement the missing command here)

**Comment by alexb231** [comment]
i actually never found the issue and it never appeared again my guess would be it was an oversight? cant really tell since theres generous wait time inbetween commands that can spend mushrooms. But im keen on hearing about the tower issue.

**Comment by the-marenga** [comment]
ok, waiting for some time probably helps. Because the issue is, that after you enter a dungeon and even if you send an update, the server is very likely to NOT send you the new timestamps for dungeons. That means you can fight the dungeon, look at the next free time, see that it is free and attack again. Resulting in an almost infinite loop, that only ends when the server sends you the updated times, if at all. For most dungeons, that is not that much of an issue, since you will receive a "fight not free" error, that you can handle, but because of prosumeably a server error, tower fights ignore the use_mushrooms flag in the command and just use a mushroom instead. Mixed with the infinite loop, that means you can loose mushrooms as fast as your check_dungeons loop is. The way to fix that is to send a `PlayerDungeonOpen` command, which is the thing, that guarantees the update times to be correct. 

Just as an rough estimate based on my error logs & guesses on how many people are using older/ the official bot versions, this bug propably costs the playerbase upwards of 10000 mushrooms per week

But yeah, just sending that command before you do your calcs, or after as a sanity check can prevent all of that

**Comment by alexb231** [comment]
Holy cow that is crazy and very good information ill need to implement this, eventough dungeons have a schedule of 15 minutes in which they check whether to even make a check this is definetely a good thing to do.

**Comment by the-marenga** [comment]
>  This bug probably exists for years now.

Yeah and there are likely a lot more. Since I implemented logging I received ~2mio error reports and ~100000 of them are errors from the game server. If I figure out anything new based on those, that could be applicable to your bot, I will let you know 😄
